lib/services/sitemaps/index.js

```diff
@@ -0,0 +1,182 @@
+            }}
+          ],
+          must_not: [
+            {term: {url: ''}}
+          ]
+        }
+      },
+      sort: [{publishTime: 'desc'}]
+    },
+  });
+
+  return h(esStream)
+    .through(closeAtLimit(esStream, URLS_PER_PAGE))
+    .invoke('toString')
+    .map(JSON.parse)
+    .map(page => ({
+      url: page.url,
+      lastmod: new Date(page.publishTime).toISOString()
+    }));
+}
+
+function closeAtLimit(esStream, limit) {
+  let counter = 0;
+
+  return stream => stream.tap(() => {
+    if (counter++ === limit - 1) esStream.close();
+  });
+}
+
+
+/**
+ * Stream documents from the sitemaps-entries index, sorted by lastmod.
+ * @param {string} site site slug
+ * @param {number} year
+ * @return {Stream}
+ */
+function streamEntriesCustom(site, year) {
+  const esStream = elastic.scrollStream({
+    index: helpers.indexWithPrefix('sitemap-entries'),
+    type: 'general',
+    scroll: '10s',
+    size: '50',
+    sort: [{lastmod: 'desc'}],
+    body: {
+      query: {
+        bool: {
+          filter: [{
+            term: {site},
+          }, {
+            range: {
+              lastmod: {
+                gte: `${year}-01-01`,
+                lte: `${year + 1}-01-01`,
+                format: 'yyyy-MM-dd'
+              }
+            }
+          }]
+        }
+      }
+    }
+  });
+
+  return h(esStream)
+    .through(closeAtLimit(esStream, URLS_PER_PAGE))
+    .invoke('toString')
+    .map(JSON.parse);
+}
+
+function streamEntries() {
+  return customSitemapExists() ?
+    streamEntriesCustom.apply(this, arguments) :
+    streamEntriesDefault.apply(this, arguments);
+}
+
+/**
+ * Stream documents from the news-sitemap-entries index, sorted by lastmod.
+ * @param {string} [site] site slug
+ * @return {Stream}
+ */
+function streamNewsEntries(site) {
+  const index = helpers.indexWithPrefix('news-sitemap-entries'),
+    esStream = elastic.scrollStream({
+      index,
+      type: 'general',
+      scroll: '10s',
+      size: '50',
+      body: {
+        sort: [{lastmod: 'desc'}],
+        query: {bool: {filter: {term: {site}}}}
+      }
+    });
+
+  return h(esStream)
+    .through(closeAtLimit(esStream, MAX_NEWS_URLS))
+    .invoke('toString')
+    .map(JSON.parse);
+}
+
+/**
+ * Return a handlebars render fnc for the specified template.
+ * @param  {string} name e.g. "foo"
+ * @return {function}
+ */
+function getRenderer(name) {
+  const filename = `${name}.handlebars`,
+    filepath = path.join(__dirname, filename),
+    s = fs.readFileSync(filepath, 'utf8');
+
+  return handlebars.compile(s);
+}
+
+/**
+ * Return true if a mapping and handler for a custom sitemap exists.
+ * @return {boolean}
+ */
+function customSitemapExists() {
+  return !!(setup.mappings['sitemap-entries'] &&
+    setup.handlers['sitemap-entries']);
+}
+
+/**
+ * Return true if a mapping and handler for a news sitemap exists.
+ * @return {boolean}
+ */
+function newsSitemapExists() {
+  return !!(setup.mappings['news-sitemap-entries'] &&
```

Comment:
If a handler subscribes to a stream then is it in the handlers object?

Reply:
Yes. A handler that doesn't export something is an empty object in `setup.handlers`